packages/wallet-ui/src/components/ui/molecule/AmountInput/AmountInput.view.tsx

```diff
@@ -112,7 +112,7 @@ export const AmountInputView = ({
   const handleMaxClick = () => {
     if (inputRef.current && asset.usdPrice) {
       const amountStr = ethers.utils
-        .formatUnits(asset.amount, asset.decimals)
+        .formatUnits(asset.spendableAmount, asset.decimals)
```

Comment:
Concerning spendableAmount. I think it can't be undefined if the asset is defined because the `updateTokenBalance` and `addTokenBalance` always call `getTokenBalanceWithDetails` which sets spendableAmount.

But I am also wondering if the check for `usdPrice` should not be removed here. Because in case we reach rate limit of coingecko then in the current situation the maximum button does network, which I am not sure is what we want. Ideally the max button should always work.

Reply:
therefore in that case, we may need to chg the type to not become optional